### PR TITLE
Add option to set owner group of daemon FIFO

### PR DESCRIPTION
**Pull request by jardous**
New option to set owner group of daemon FIFO (Default: /tmp/dlt)
is added in dlt.conf. If this option is used properly, more
secure tracing can be realized.

Only application that is in dlt_user_apps_group can write log
message to daemon FIFO.

### CODE REVIEW

File: src/daemon/dlt.conf [hunk 1]
```diff
@@ -81,6 +81,56 @@ RingbufferStepSize = 500000
 # Read gateway configuration from another location
 # GatewayConfigFile = /etc/dlt_gateway.conf
 
+
+########################################################################
```
Comment (ssugiura) [inline src/daemon/dlt.conf]: These explanations are too long for dlt.conf and we don't need hypothesis (or just summarize it in commit message). Please have a short description for this and add it to dlt.conf.md under doc instead.

File: src/daemon/dlt-daemon.h [hunk 2]
```diff
@@ -125,7 +125,8 @@ typedef struct
     char userPipesDir[NAME_MAX + 1]; /**< (String: Directory) directory where dltpipes reside (Default: /tmp/dltpipes) */
 #endif
     char daemonFifoName[NAME_MAX + 1]; /**< (String: Filename) name of local fifo (Default: /tmp/dlt) */
-    unsigned int port;  /**< port number */
+    char daemonFifoGroup[NAME_MAX + 1]; /**< (String: Group name) Owner group of local fifo (Default: Primary Group) */
+    unsigned int  port; /**< port number */
```
Comment (ssugiura) [inline src/daemon/dlt-daemon.h]: Extra space before port

File: src/daemon/dlt-daemon.c [hunk 3]
```diff
@@ -1048,6 +1055,49 @@ static int dlt_daemon_init_fifo(DltDaemonLocal *daemon_local)
...
+        {
+            ret = chown(tmpFifo, -1, group_dlt->gr_gid);
+            if (ret == -1)
+            {
+                snprintf(str, DLT_DAEMON_TEXTBUFSIZE,
+                         "FIFO user %s cannot be chowned to group %s (%s)\n",
+                         tmpFifo,
+                         daemon_local->flags.daemonFifoGroup,
+                         strerror(errno));
+                dlt_log(LOG_ERR, str);
```
[the remark this answers is not in the document]
Comment (jardous) [reply src/daemon/dlt-daemon.c]: done

File: src/daemon/dlt-daemon.c [hunk 4]
```diff
@@ -1048,6 +1055,49 @@ static int dlt_daemon_init_fifo(DltDaemonLocal *daemon_local)
...
+            }
+        }
+        else if (errno == 0)
+        {
+            snprintf(str, DLT_DAEMON_TEXTBUFSIZE,
+                     "Group name %s is not found\n",
+                     daemon_local->flags.daemonFifoGroup);
+            dlt_log(LOG_ERR, str);
+        }
+        else if ((errno == ENOENT) || (errno == EBADF) || (errno == EPERM))
```
[the remark this answers is not in the document]
Comment (jardous) [reply src/daemon/dlt-daemon.c]: done

File: src/daemon/dlt.conf (hunk 1 above)
Comment (jardous) [reply src/daemon/dlt.conf]: done

File: src/daemon/dlt-daemon.h (hunk 2 above)
Comment (jardous) [reply src/daemon/dlt-daemon.h]: done

**Pull request closed without merging**